pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/JavaTypeDefinitionSimple.java

```diff
@@ -83,12 +80,25 @@ protected JavaTypeDefinitionSimple(Class<?> clazz, JavaTypeDefinition... boundGe
 
     @Override
     public JavaTypeDefinition getEnclosingClass() {
-        return enclosingClass;
+        return JavaTypeDefinition.forClass(loadEnclosing(clazz));
+    }
+
+    @Override
+    public boolean isRawType() {
+        return isGeneric() && typeArgumentCount == 0;
     }
 
     @Override
     public boolean isGeneric() {
-        return isGeneric;
+        return getTypeParameterCount() != 0;
+    }
+
+    @Override
+    public int getTypeParameterCount() {
+        if (typeParameterCount == -1) {
+            typeParameterCount = getTypeParameters(clazz).length;
```

Comment:
I think, we should add here a try-catch for LinkageError. In case, we couldn't determine the type parameters, can we then set typeParametersCount=0 ?

Reply:
I think you're right, will do